From rsync-admin@lists.samba.org Wed Mar 13 17:38:07 2002
Return-Path: <rsync-admin@lists.samba.org>
Delivered-To: mbp@samba.org
Received: from va.samba.org (localhost [127.0.0.1])
	by lists.samba.org (Postfix) with ESMTP
	id 3CE014420; Wed, 13 Mar 2002 17:38:06 -0800 (PST)
Delivered-To: rsync@lists.samba.org
Received: from mercury.Sun.COM (mercury.Sun.COM [192.9.25.1])
	by lists.samba.org (Postfix) with ESMTP id 3C43E43FE
	for <rsync@lists.samba.org>; Wed, 13 Mar 2002 17:37:20 -0800 (PST)
Received: from engmail3.Eng.Sun.COM ([129.144.170.5])
	by mercury.Sun.COM (8.9.3+Sun/8.9.3) with ESMTP id RAA08864;
	Wed, 13 Mar 2002 17:37:14 -0800 (PST)
Received: from goodluck.Eng.Sun.COM (goodluck.Eng.Sun.COM [129.144.28.81])
	by engmail3.Eng.Sun.COM (8.9.3+Sun/8.9.3/ENSMAIL,v2.1p1) with ESMTP id RAA25195;
	Wed, 13 Mar 2002 17:37:14 -0800 (PST)
Received: from goodluck (goodluck [129.144.28.81])
	by goodluck.Eng.Sun.COM (8.9.3+Sun/8.9.1) with SMTP id RAA05041;
	Wed, 13 Mar 2002 17:37:13 -0800 (PST)
Message-Id: <200203140137.RAA05041@goodluck.Eng.Sun.COM>
From: Jennifer Lu <mou@goodluck.Eng.Sun.COM>
Reply-To: Jennifer Lu <mou@goodluck.Eng.Sun.COM>
Subject: Re: map_ptr warning
To: rsync@lists.samba.org, cmount@nortelnetworks.com
MIME-Version: 1.0
Content-Type: TEXT/plain; charset=us-ascii
Content-MD5: pjY7k5RvgQaUX4EBLt+diQ==
X-Mailer: dtmail 1.3.0 @(#)CDE Version 1.3.4 SunOS 5.7 sun4u sparc 
Sender: rsync-admin@lists.samba.org
Errors-To: rsync-admin@lists.samba.org
X-BeenThere: rsync@lists.samba.org
X-Mailman-Version: 2.0.8
Precedence: bulk
List-Help: <mailto:rsync-request@lists.samba.org?subject=help>
List-Post: <mailto:rsync@lists.samba.org>
List-Subscribe: <http://lists.samba.org/mailman/listinfo/rsync>,
	<mailto:rsync-request@lists.samba.org?subject=subscribe>
List-Id: rsync user list <rsync.lists.samba.org>
List-Unsubscribe: <http://lists.samba.org/mailman/listinfo/rsync>,
	<mailto:rsync-request@lists.samba.org?subject=unsubscribe>
List-Archive: <http://lists.samba.org/pipermail/rsync/>
Date: Wed, 13 Mar 2002 17:37:13 -0800 (PST)
Status: RO
Content-Length: 2455
Lines: 69



Charles,


> Delivered-To: rsync@lists.samba.org
> From: "Charles Mount" <cmount@nortelnetworks.com>
> To: rsync@lists.samba.org
> Subject: map_ptr warning
> MIME-Version: 1.0
> X-BeenThere: rsync@lists.samba.org
> X-Mailman-Version: 2.0.8
> List-Help: <mailto:rsync-request@lists.samba.org?subject=help>
> List-Post: <mailto:rsync@lists.samba.org>
> List-Subscribe: <http://lists.samba.org/mailman/listinfo/rsync>, 
<mailto:rsync-request@lists.samba.org?subject=subscribe>
> List-Id: rsync user list <rsync.lists.samba.org>
> List-Unsubscribe: <http://lists.samba.org/mailman/listinfo/rsync>, 
<mailto:rsync-request@lists.samba.org?subject=unsubscribe>
> List-Archive: <http://lists.samba.org/pipermail/rsync/>
> Date: Wed, 13 Mar 2002 15:50:06 -0600
> 
> I am using rsync-2.5.2.Solaris251.sparc from your site to synchronize from
> an old  Solaris2.5.1 server in Nashville to a new Solaris8 server in
> Raleigh.   I will air express tapes and use rsync to update.  I am dealing
> with several hundred thousand files and a couple hundred gigabytes of data.
> I use "rsync -avx --delete -e ssh old:/foo/bar/ /foo/bar >>2
> /var/log/rsynclog" run as a cronjob on the new computer.
> Most of the files are transferred correctly.
> With a few files I get the file name followed the two copies of the line:
> "Warning: unexpected read size of 0 in map_ptr"  in the log file.
> In at least most cases, the next pass gives the same message on the same
> file.  The file size is sometimes larger on the target and may grow on
> additional passes.  If I delete the file on the target the next pass will
> normally work correctly.  I see no pattern of where the problem will strike.
> In some cases, the problem files are plain text with simple path and file
> names.
> Any ideas?

I got the same errors. Upgrade rsync to 2.5.3 with updating batch.c file 

diff -u -r1.13 batch.c
--- batch.c     6 Feb 2002 21:20:48 -0000   1.13
+++ batch.c     12 Mar 2002 06:16:49 -0000
@@ -396,7 +396,7 @@
                   struct sum_struct *s)
 {
     size_t i;
-    unsigned int int_zero = 0;
+    size_t int_zero = 0;
     extern int csum_length;
 
     fdb_open = 1;
     
Then, re-compile it. After I compiled the new version of rsync, it works perfectly.

     
> 
> Thanks,
> Charles Mount
> 
> 

Jennifer


-- 
To unsubscribe or change options: http://lists.samba.org/mailman/listinfo/rsync
Before posting, read: http://www.tuxedo.org/~esr/faqs/smart-questions.html

